subsys/bluetooth/mesh/proxy_srv.c

```diff
@@ -585,29 +586,44 @@ static int gatt_proxy_advertise(struct bt_mesh_subnet *sub)
 		}
 	}
 
-	if (sub->node_id == BT_MESH_NODE_IDENTITY_RUNNING) {
-		uint32_t active = k_uptime_get_32() - sub->node_id_start;
+	for (int i = 0; i < subnet_count; i++) {
 
-		if (active < NODE_ID_TIMEOUT) {
-			remaining = NODE_ID_TIMEOUT - active;
-			BT_DBG("Node ID active for %u ms, %d ms remaining",
-			       active, remaining);
-			err = node_id_adv(sub, remaining);
-		} else {
-			bt_mesh_proxy_identity_stop(sub);
-			BT_DBG("Node ID stopped");
+		if (sub->node_id == BT_MESH_NODE_IDENTITY_RUNNING) {
+			uint32_t active = k_uptime_get_32() - sub->node_id_start;
+
+			if (active < NODE_ID_TIMEOUT) {
+				remaining = NODE_ID_TIMEOUT - active;
+				BT_DBG("Node ID active for %u ms, %d ms remaining",
+				       active, remaining);
+				err = node_id_adv(sub, remaining);
+				planned = true;
+			} else {
+				bt_mesh_proxy_identity_stop(sub);
+				BT_DBG("Node ID stopped");
+			}
 		}
-	}
 
-	if (sub->node_id == BT_MESH_NODE_IDENTITY_STOPPED) {
-		err = net_id_adv(sub, remaining);
-	}
+		/* Mesh Profile Specification v1.0.1, section 7.2.2.2.1
+		 * A node that does not support the Proxy feature or
+		 * has the GATT Proxy state disabled shall not advertise with Network ID.
+		 */
+		if (sub->node_id == BT_MESH_NODE_IDENTITY_STOPPED &&
+			bt_mesh_gatt_proxy_get() == BT_MESH_FEATURE_ENABLED) {
+			err = net_id_adv(sub, remaining);
```

Comment:
This'll either send a net ID or a node ID for each subnet, then exit. Is this the intention? Or should it check whether node ID is ready to run in **any** subnet, _then_ fall back to net ID?

Reply:
Hm. The adv will be planned only once and then the loop is left on line 620. No more than one adv is planned at a time.
No more than one cycle is done for the loop. `beacon_sub` points always on the first nor planned subnetwork yet.  
I guess it prevents the start of the already started adv.

Reply:
Right. I had to re-read this function a couple of time to get it. It has been refactored to death (before this PR), and a lot of the code above this loop is nonsensical. I can make a followup PR after this one to clean that up.